lib/langchain/llm/openai.rb

```diff
@@ -116,19 +118,20 @@ def chat(prompt: "", messages: [], context: "", examples: [], **options)
       parameters = compose_parameters @defaults[:chat_completion_model_name], options
       parameters[:messages] = compose_chat_messages(prompt: prompt, messages: messages, context: context, examples: examples)
       parameters[:max_tokens] = validate_max_tokens(parameters[:messages], parameters[:model])
-
+      parameters[:functions] = @functions unless @functions.nil?
       if (streaming = block_given?)
         parameters[:stream] = proc do |chunk, _bytesize|
-          yield chunk.dig("choices", 0, "delta", "content")
+          yield chunk if @complete_chunk
+          yield chunk.dig("choices", 0, "delta", "content") unless @complete_chunk
         end
       end
 
+      parameters.delete(:max_tokens) if @functions
```

Comment:
Instead of deleting it here, can we just never set it in the first place? A few lines above:
```ruby
parameters[:max_tokens] = validate_max_tokens(parameters[:messages], parameters[:model]) unless functions
```